src/qibo/backends/qulacs.py

```diff
@@ -0,0 +1,87 @@
+import re
+
+import numpy as np
+import qulacs  # pylint: disable=import-error
+from qulacs import (  # pylint: disable=no-name-in-module, import-error
+    QuantumCircuitSimulator,
+    converter,
+)
+
+from qibo import __version__
+from qibo.backends import NumpyBackend
+from qibo.config import raise_error
+from qibo.result import CircuitResult, QuantumState
+
+
+def circuit_to_qulacs(
+    circuit: "qibo.Circuit",
+) -> "qulacs.QuantumCircuit":  # pylint: disable=no-member
+    """
+    Converts a qibo circuit in a qulacs circuit.
+
+    Args:
+        circuit (:class:`qibo.models.circuit.Circuit`): Input circuit to convert.
+
+    Returns:
+        qulacs.QuantumCircuit: The converted qulacs circuit.
+    """
+    qasm_str = re.sub("^//.+\n", "", circuit.to_qasm())
+    qasm_str = re.sub(r"creg\s.+;", "", qasm_str)
+    qasm_str = re.sub(r"measure\s.+;", "", qasm_str)
+    circ = converter.convert_QASM_to_qulacs_circuit(qasm_str.splitlines())
+    return circ
+
+
+class QulacsBackend(NumpyBackend):
+
+    def __init__(self):
+        super().__init__()
+
+        self.qulacs = qulacs
```

Comment:
Sorry to be annoying: the motivation for `self.np` is that there is a (non-standard) NumPy interface adopted even by some other packages, so you could hope to swap `self.np = numpy` with `self.np = jax.numpy` and hope it will work.

However, I don't see much the point of `self.qulacs`...

Reply:
At the moment, there isn't, I just thought that in the future maybe one might need to call `qulacs` primitives from the backend in files where `qulacs` is not explicitely imported.

Reply:
Well, if you can import `qulacs` somewhere, you can import it everywhere. That's mainly up to Python's module system, but it already guarantees that it will always be the same `qulacs` (unless explicitly unloaded)